lib/model/Product.dart

```diff
@@ -1,188 +1,274 @@
+  static const String _JSON_TAG_NUTRITION_DATA_PER = 'nutrition_data_per';
+  static const String _JSON_TAG_NUTRITION_GRADE_FR = 'nutrition_grade_fr';
+  static const String _JSON_TAG_NUTRIENTS = 'nutriments';
+  static const String _JSON_TAG_ADDITIVES_TAGS = 'additives_tags';
+  static const String _JSON_TAG_NUTRIENTS_LEVELS = 'nutrient_levels';
+  static const String _JSON_TAG_SERVING_SIZE = 'serving_size';
+  static const String _JSON_TAG_SERVING_QUANTITY = 'serving_quantity';
+  static const String _JSON_TAG_BRANDS_TAGS = 'brands_tags';
+  static const String _JSON_TAG_PRODUCT_QUANTITY = 'product_quantity';
+  static const String _JSON_TAG_SELECTED_IMAGES = 'selected_images';
+  static const String _JSON_TAG_IMAGES = 'images';
+  static const String _JSON_TAG_INGREDIENTS = 'ingredients';
+  static const String _JSON_TAG_ENVIRONMENT_IMPACT_LEVELS =
+      'environment_impact_level_tags';
+  static const String _JSON_TAG_ALLERGENS_TAGS = 'allergens_tags';
+  static const String _JSON_TAG_INGREDIENTS_ANALYSIS_TAGS =
+      'ingredients_analysis_tags';
+  static const String _JSON_TAG_CATEGORIES_TAGS = 'categories_tags';
+  static const String _JSON_TAG_CATEGORIES_TAGS_TRANSLATED =
+      'categories_tags_translated';
+  static const String _JSON_TAG_LABELS_TAGS = 'labels_tags';
+  static const String _JSON_TAG_LABELS_TAGS_TRANSLATED =
+      'labels_tags_translated';
+  static const String _JSON_TAG_MISC = 'misc';
+  static const String _JSON_TAG_STATES_TAGS = 'states_tags';
+  static const String _JSON_TAG_TRACES_TAGS = 'traces_tags';
+  static const String _JSON_TAG_STORES_TAGS = 'stores_tags';
+  static const String _JSON_TAG_ATTRIBUTE_GROUPS = 'attribute_groups';
+  static const String _JSON_TAG_LAST_MODIFIED = 'last_modified_t';
+  static const String _JSON_TAG_ECOSCORE_GRADE = 'ecoscore_grade';
 
-  @JsonKey(name: 'ingredients_text', includeIfNull: false)
-  String ingredientsText;
-  @JsonKey(name: 'ingredients_text_de', includeIfNull: false)
-  String ingredientsTextDE;
-  @JsonKey(name: 'ingredients_text_en', includeIfNull: false)
-  String ingredientsTextEN;
-  @JsonKey(name: 'ingredients_text_fr', includeIfNull: false)
-  String ingredientsTextFR;
-
-  @JsonKey(
-      name: 'ingredients_analysis_tags',
-      includeIfNull: false,
-      fromJson: IngredientsAnalysisTags.fromJson,
-      toJson: IngredientsAnalysisTags.toJson)
-  IngredientsAnalysisTags ingredientsAnalysisTags;
-
-  @JsonKey(name: 'nutriment_energy_unit', includeIfNull: false)
-  String nutrimentEnergyUnit;
-  @JsonKey(name: 'nutrition_data_per', includeIfNull: false)
-  String nutrimentDataPer;
-  @JsonKey(name: 'nutrition_grade_fr', includeIfNull: false)
-  String nutriscore;
-
-  @JsonKey(includeIfNull: false)
-  String categories;
-
-  @JsonKey(name: 'categories_tags', includeIfNull: false)
-  List<String> categoriesTags;
-  @JsonKey(name: 'categories_tags_translated', includeIfNull: false)
+  final String barcode;
```

Comment:
After spending some more time testing this PR with my app I'm getting several compilation errors because fields have been changed to final, which breaks compatibility. I rely a lot on this in my app tests, i.e. instantiating a `Product`, setting some fields, performing some tests, then updating some additional fields for the same object and running some more tests. I think the `final` fields should be avoided.